src/components/Profile/PostEdit.jsx

```diff
@@ -0,0 +1,43 @@
+import axios from "axios"
+import { useModal, usePosts } from "contexts"
+import { useEffect, useState } from "react"
+import { ACTION_EDIT_POST } from "utils"
+import styles from './profile.module.css'
+
+const PostEdit = () => {
+    const [post, setPost] = useState({})
+    const { modal: { id }, setModal } = useModal()
+    const { editPost, postsDispatch } = usePosts()
+
+    const handlePostEdit = async () => {
+        const response = await editPost(id, post)
+        if (response.status === 201) {
+            const editedPost = response.data.posts.find(newpost => newpost._id === post._id)
+            postsDispatch({ type: ACTION_EDIT_POST, payload: editedPost })
+        }
+        setPost({})
+        setModal(m => ({ ...m, type: '', id: '' }))
```

Comment:
suggestion: try to be more descriptive with var naming.

Reply:
Sure, I'll make the changes.